Tests/FluentAssertions.Net40.Specs/XmlNodeAssertionSpecs.cs

```diff
@@ -0,0 +1,489 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml;
+
+namespace FluentAssertions.Specs
+{
+    [TestClass]
+    public class XmlNodeAssertionSpecs
+    {
+        #region Be / NotBe
+
+        [TestMethod]
+        public void When_asserting_an_XmlNode_is_the_same_as_the_same_XmlNode_it_should_succeed()
+        {
+            //-------------------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-------------------------------------------------------------------------------------------------------------------
+            var doc = new XmlDocument();
+
+            //-------------------------------------------------------------------------------------------------------------------
+            // Act
+            //-------------------------------------------------------------------------------------------------------------------
+            Action act = () =>
+                doc.Should().BeSameAs(doc);
+
+            //-------------------------------------------------------------------------------------------------------------------
+            // Assert
+            //-------------------------------------------------------------------------------------------------------------------
+            act.ShouldNotThrow();
+        }
+
+        [TestMethod]
+        public void When_asserting_an_XmlDocument_is_same_as_a_different_XmlDocument_it_should_fail_with_descriptive_message()
+        {
+            //-------------------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-------------------------------------------------------------------------------------------------------------------
+            var doc = new XmlDocument();
+            doc.LoadXml("<doc/>");
+            var otherNode = new XmlDocument();
+            otherNode.LoadXml("<otherDoc/>");
+
+            //-------------------------------------------------------------------------------------------------------------------
+            // Act
+            //-------------------------------------------------------------------------------------------------------------------
+            Action act = () =>
+                doc.Should().BeSameAs(otherNode, "we want to test the failure {0}", "message");
+
+            //-------------------------------------------------------------------------------------------------------------------
+            // Assert
+            //-------------------------------------------------------------------------------------------------------------------
+            act.ShouldThrow<AssertFailedException>()
+                .WithMessage("Expected object to refer to <otherDoc /> because we want to test the failure message, but found object <doc />.");
```

Comment:
Now I finally think I have found the reason:

Looking at `ReferenceTypeAssertion<>.BeNull()` the failure message is formatted with:

````
.FailWith("Expected {context:" + Context + "} to be <null>{reason}, but found {0}.", Subject);
````

It uses the `Context` property. But looking at ReferenceTypeAssertion<>.BeSameAs()` the failure message is formatted with:

````
.FailWith("Expected {context:object} to refer to {0}{reason}, but found object {1}.", expected, Subject);
````

Shouldn't that be changed to pick up the name from `Context` instead of hard coding it to "object"?

````
.FailWith("Expected {context:" + Context + "} to refer to {0}{reason}, but found object {1}.", expected, Subject);
````

If that change is done, I've already overloaded `Context` in the `XmlNodeAssertions` class, so it will be presented as "XmlNode".